helpers-json/json.test.js

```diff
@@ -731,6 +731,163 @@ suite("Parse illegally structured ARB", () => {
+    });
+
+    test("translateResource removes deeply nested placeholder annotations when emitArbAnnotations is false", async () => {
+        const resourceFilter = new i18next.I18nextFilter({
+            enableArbAnnotations: true,
+            emitArbAnnotations: false,
+        });
+        const translator = async (sid, str) => `${str} - **Translated**`;
+        const resource = {
+            recentSearch: {
+                cars: {
+                    differentPlaceDropOff: "Drop off at {{pickupLocationName}} instead of pickup location",
+                    "@differentPlaceDropOff": {
+                        description: "Message when drop-off differs from pickup for car rentals",
+                        placeholders: {
+                            pickupLocationName: {
+                                example: "Downtown San Francisco",
+                                description: "The name of the pickup location for car rental"
+                            }
+                        }
+                    }
+                }
+            }
+        };
+        const expectedOutput = {
+            recentSearch: {
+                cars: {
+                    differentPlaceDropOff: "Drop off at {{pickupLocationName}} instead of pickup location - **Translated**"
+                }
+            }
+        };
+        const output = await resourceFilter.translateResource({
+            resource: JSON.stringify(resource),
+            translator,
+        });
+        assert.deepEqual(JSON.parse(output), expectedOutput);
+    });
+
+    test("translateResource removes nested annotations when translation is null", async () => {
+        const resourceFilter = new i18next.I18nextFilter({
+            enableArbAnnotations: true,
+            emitArbAnnotations: true,
+        });
+        const translator = async (sid, str) => {
+            // Return null for the nested key to test annotation removal
+            return sid === "recentSearch.cars.differentPlaceDropOff" ? null : `${str} - **Translated**`;
+        };
+        const resource = {
+            recentSearch: {
+                cars: {
+                    differentPlaceDropOff: "Drop off at {{pickupLocationName}} instead of pickup location",
+                    "@differentPlaceDropOff": {
+                        description: "Message when drop-off differs from pickup for car rentals",
+                        placeholders: {
+                            pickupLocationName: {
+                                example: "Downtown San Francisco",
+                                description: "The name of the pickup location for car rental"
+                            }
+                        }
+                    },
+                    anotherKey: "Another message"
+                }
+            }
+        };
+        const expectedOutput = {
+            recentSearch: {
+                cars: {
+                    differentPlaceDropOff: null,
```

Comment:
when a translation is `null` it means it's missing. do we really want to produce `null` in the translated file? shouldn't the property be either missing or fall back to english? (i assume i18next handles the fallbacks but does it work when the json says `null` rather than missing?)

Reply:
Good catch.  I modified both the test and the condition to catch missing rather than undefined.